### PR TITLE
fix redirection out of scope bugs

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#3326

File: src/docfx/Errors.cs [hunk 1]
```diff
@@ -85,6 +85,9 @@ public static Error LinkIsEmpty(Document relativeTo)
         public static Error LinkOutOfScope(Document relativeTo, Document file, string href)
             => new Error(ErrorLevel.Warning, "link-out-of-scope", $"File '{file}' referenced by link '{href}' will not be build because it is not included in docfx.yml", relativeTo.ToString());
 
+        public static Error RedirectionOutOfScope(Document redirection)
+            => new Error(ErrorLevel.Warning, "redirection-out-of-scope", $"Redirection file '{redirection}' will not be build because it is not included in docfx.yml");
```
Comment (live1206) [inline src/docfx/Errors.cs]: >build [](start = 120, length = 5)

should be "built"

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -85,6 +85,9 @@ public static Error LinkIsEmpty(Document relativeTo)
         public static Error LinkOutOfScope(Document relativeTo, Document file, string href)
             => new Error(ErrorLevel.Warning, "link-out-of-scope", $"File '{file}' referenced by link '{href}' will not be build because it is not included in docfx.yml", relativeTo.ToString());
```
Comment (live1206) [inline src/docfx/Errors.cs]: >build [](start = 122, length = 5)

built

File: src/docfx/Errors.cs [hunk 3]
```diff
@@ -85,6 +85,9 @@ public static Error LinkIsEmpty(Document relativeTo)
...
 
+        public static Error RedirectionOutOfScope(Document redirection)
+            => new Error(ErrorLevel.Warning, "redirection-out-of-scope", $"Redirection file '{redirection}' will not be build because it is not included in docfx.yml");
+
         public static Error LinkIsDependency(Document relativeTo, Document file, string href)
             => new Error(ErrorLevel.Warning, "link-is-dependency", $"File '{file}' referenced by link '{href}' will not be build because it is from a dependency docset", relativeTo.ToString());
```
Comment (live1206) [inline src/docfx/Errors.cs]: >build [](start = 123, length = 5)

built

**Pull request merged**